server/game/cards/22-BtB/Highgarden.js

```diff
@@ -0,0 +1,140 @@
+const DrawCard = require('../../drawcard.js');
+const GameActions = require('../../GameActions');
+const Message = require('../../Message');
+const range = require('lodash.range');
+
+class Highgarden extends DrawCard {
+    setupCardAbilities(ability) {
+        this.action({
+            title: 'Give characters +STR',
+            condition: () => this.controller.hand.length > 0,
+            cost: ability.costs.kneelSelf(),
+            handler: () => {
+                this.game.promptForSelect(this.controller, {
+                    activePromptTitle: 'Reveal up to 3 cards',
+                    source: this,
+                    numCards: 3,
+                    mode: 'upTo',
+                    cardCondition: card => card.controller === this.controller && card.location === 'hand' && card.isFaction('tyrell') && card.getType() === 'character',
+                    onSelect: (player, cards) => this.revealSelect(player, cards),
+                    onCancel: (player) => this.revealCancel(player)
+                });
```

Comment:
Would it make sense to change this to a `target` declaration with `mode: 'select'` (since it doesn't have the word choose).  This would have the benefits of:
1. The player can't trigger the ability unless they actually intend to reveal 1 or more cards (unless revealing 0 cards is intentionally allowed by the rules, in which case the current cancel message should not be a `danger` message)
2. Can get rid of the `revealCancel` method and fold the `revealSelect` into the handler.
3. Easier set up for using GameActions in the future.